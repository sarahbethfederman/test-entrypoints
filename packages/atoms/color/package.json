{
  "name": "@auscred/color",
  "version": "0.0.1",
  "source": "src/Color.tsx",
  "main": "dist/Colour.js",
  "types": "dist/Colour.d.js",
  "files": [
    "lib"
  ],
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "peerDependencies": {
    "@types/react": "^16.4.0",
    "react": "^16.4.0",
    "styled-components": "^3.3.3"
  },
  "scripts": {
    "build": "tsc --pretty"
  }
}
  