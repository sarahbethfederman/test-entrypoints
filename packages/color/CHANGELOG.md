2.0.2 - New Changelog
