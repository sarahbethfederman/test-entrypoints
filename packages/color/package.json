{
  "name": "@auscred/color",
  "version": "1.0.0",
  "source": "src/index.ts",
  "main": "dist/index.js",
  "types": "dist/index.d.js",
  "files": [
    "dist"
  ],
  "license": "ISC",
  "peerDependencies": {
    "@auscred/theme": "^1.0.0",
    "styled-components": "^3.3.3"
  },
  "dependencies": {
    "@auscred/breakpoint": "^1.0.0"
  },
  "scripts": {
    "build": "tsc"
  }
}
  