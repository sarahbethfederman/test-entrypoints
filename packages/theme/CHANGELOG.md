## [9.1.0] 2020-06-29 [FUNNEL-1473](https://creditandfinance.atlassian.net/browse/FUNNEL-1473)

## 10.0.0-beta.0

### Minor Changes

- preconstruct infra update

### Patch Changes

- Updated dependencies [undefined]
  - @lendi-ui/breakpoint@6.0.0-beta.0
- Change `--lend-ui-size` to use new t-shirt breakpoints.

## [9.0.0] 2020-06-29

### Changed

- lendi-ui-size root size breakpoint changed to 576px instead of 577px

## [8.1.0] 2019-10-02 [HUB-589](https://creditandfinance.atlassian.net/browse/HUB-589)

### Changed

- Enhancement LUI theme with default font-family for body and font-weight for heading.

## [8.0.1] 2019-9-29 [HUB-589](https://creditandfinance.atlassian.net/browse/HUB-589)

### Changed

- added font family defaults based on theme

## [8.0.0] 2019-9-23 - 2019-08-15 - [HUB-412](https://creditandfinance.atlassian.net/browse/HUB-412)

## [7.0.0] 2019-9-23 - 2019-08-15 - [HUB-412](https://creditandfinance.atlassian.net/browse/HUB-412)

### Changed

- Color is now a type
- ColorPrefix is now a type
- theme.color object has been flattened
- Update typescript to version 3.5.2 and react to version 16.8.4

## [6.2.0] - 2019-05-17 - [HUB-174](https://creditandfinance.atlassian.net/browse/HUB-174)

### Changed

- Module format reverted back to CJS.

### Deleted

- Dropped "module" property from package.json.

## [6.1.1] - 2019-05-10 - [HUB-137](https://creditandfinance.atlassian.net/browse/HUB-137)

### Added

- Add `-ms-clear` into theme injectGlobal to prevent cross icon in input by Edge and IE browser default style

## [6.1.0] - 2019-05-02 - [HUB-88](https://creditandfinance.atlassian.net/browse/HUB-88)

### Changed

- replace `border-radius: 6px` with select function in theme

## [6.0.2] - 2019-05-09 - [HUB-92](https://creditandfinance.atlassian.net/browse/HUB-92)

### Changed

- Updated CHANGELOG to match required parsing format.

## [6.0.1] - 2019-05-08 - [HUB-54](https://creditandfinance.atlassian.net/browse/HUB-54)

### Changed

- Change module main entry point to esm in package.json

## [6.0.0] - 2019-04-30 - [HUB-54](https://creditandfinance.atlassian.net/browse/HUB-54)

### Deleted

- Remove the CJS bundles.

## [5.0.0] - 2019-03-06 - [BRAVO-387](https://creditandfinance.atlassian.net/browse/BRAVO-387)

### Changed

- Added optional theme kind prop.

## [4.0.2] - 2019-02-25 - [BRAVO-511](https://creditandfinance.atlassian.net/browse/BRAVO-511)

### Changed

- Change warning colour to meet accessibility standards.

## [4.0.1]

### Added

- Added CHANGELOG file.
