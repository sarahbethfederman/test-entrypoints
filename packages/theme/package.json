{
  "name": "@lendi-ui/theme",
  "version": "9.1.0",
  "source": "src/index.ts",
  "main": "dist/cjs/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist"
  ],
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "peerDependencies": {
    "@lendi-ui/breakpoint": "^5.0.0",
    "@types/react": "^16.8",
    "@types/styled-components": "^4.1.18",
    "react": "^16.8",
    "styled-components": "^4"
  },
  "scripts": {
    "build:cjs": "tsc --project tsconfig.cjs.json",
    "build:types": "tsc --project tsconfig.types.json",
    "build": "yarn run build:types && yarn run build:cjs"
  }
}
