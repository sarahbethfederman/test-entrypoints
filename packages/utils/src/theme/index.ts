export type ColorKeys = '25' | '50' | '100' | '200' | '300' | '400' | '500' | '600' | '700' | '800' | '900' | '950';

export type ColorsMap = { [key in ColorKeys]: string };

export type Typography = {
  heading: {
    fontFamily: string;
  };

  body: {
    fontFamily: string;
  };
};

export type Logo = {
  logoName: string;
};

export type Colors = {
  primary: ColorsMap;
  secondary: ColorsMap;
  tertiary: ColorsMap;
  shade: ColorsMap;
  info: ColorsMap;
  warn: ColorsMap;
  error: ColorsMap;
  success: ColorsMap;
};

export type ThemeMap = {
  typography: Typography;
  colors: Colors;
  logo: Logo;
};
