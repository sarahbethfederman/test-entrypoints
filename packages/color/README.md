# @lendi-ui/color

Color style with `bg`(background color), `fg`(font color) and `color`(every CSS property relate to color).

## Installation

```
yarn add @lendi-ui/color
```

## Usage

- `bg` for styling background-color.

```jsx
import { bg } from '@lendi-ui/color';
const Square = styled.div`
  width: 2em;
  height: 2em;
  ${bg('warn.500')};
`;
```

- `fg` for styling font-color.

```jsx
import { fg } from '@lendi-ui/color';
const WarningText = styled.span`
  ${fg('warn.500')};
`;
```

- `color` for styling every CSS porperty relate to color.

```jsx
import { color } from '@lendi-ui/color';
const Circle = styled.div`
  width: 0;
  border-radius: 1em;
  border: 1em solid ${color('primary.500')};
`;
```
