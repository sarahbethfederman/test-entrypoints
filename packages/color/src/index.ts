import { css } from 'styled-components';
import { select } from '@lendi-ui/theme';
import { map, BreakpointValue, BreakpointValueMap } from '@lendi-ui/breakpoint';

export type NameOrNameMap = BreakpointValue<string> | BreakpointValueMap<string>;

export function color(name: NameOrNameMap) {
  return select(`colors.${name}`);
}

export function fg(name: NameOrNameMap) {
  return map(
    name,
    (n?: string) =>
      n === undefined
        ? ''
        : css`
            color: ${color(n)};
          `
  );
}

export function bg(name: NameOrNameMap) {
  return map(
    name,
    (n?: string) =>
      n === undefined
        ? ''
        : css`
            background-color: ${color(n)};
          `
  );
}
