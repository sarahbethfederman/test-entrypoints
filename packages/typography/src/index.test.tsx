import * as React from 'react';
import styled from 'styled-components';
import { shallow } from 'enzyme';
import { theme } from '@lendi-ui/theme';
import { color } from '@lendi-ui/color';
import { heading, body } from './index';

describe('heading()', () => {
  const headingSizes = ['xl', 'lg', 'md', 'sm', 'xs'];
  const headingColors = ['primary.500', 'shade.0', 'shade.1000'];
  const headingAlignments = ['left', 'center', 'right', 'justify'];

  headingSizes.forEach((headingSize) => {
    it(`should display styles for size ${headingSize}`, () => {
      const Component = styled.div`
        ${heading({ size: headingSize })};
      `;
      const wrapper = shallow(<Component theme={theme} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  headingColors.forEach((headingColor) => {
    it(`should display styles for color ${headingColor}`, () => {
      const Component = styled.div`
        ${heading({ size: 'xl', color: headingColor })};
      `;
      const wrapper = shallow(<Component theme={theme} />);
      expect(wrapper).toHaveStyleRule('color', color(headingColor)({ theme }));
    });
  });

  headingAlignments.forEach((headingAlignment) => {
    it(`should display styles for align ${headingAlignment}`, () => {
      const Component = styled.div`
        ${heading({ size: 'xl', align: headingAlignment })};
      `;
      const wrapper = shallow(<Component theme={theme} />);
      expect(wrapper).toHaveStyleRule('text-align', headingAlignment);
    });
  });

  it(`display styles for the default color`, () => {
    const Component = styled.div`
      ${heading({ size: 'xl' })};
    `;
    const wrapper = shallow(<Component theme={theme} />);
    expect(wrapper).toHaveStyleRule('color', color('shade.700')({ theme }));
  });

  it(`should display styles for the default align`, () => {
    const Component = styled.div`
      ${heading({ size: 'xl' })};
    `;
    const wrapper = shallow(<Component theme={theme} />);
    expect(wrapper).not.toHaveStyleRule('text-align');
  });
});
