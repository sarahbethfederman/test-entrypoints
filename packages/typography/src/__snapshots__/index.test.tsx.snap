// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`heading() should display styles for size 1 1`] = `
.c0 {
  font-size: 46px;
  line-height: 1.2173913043478262;
  font-weight: bold;
  font-family: 'Cabin',sans-serif;
}

<div
  className="c0"
/>
`;

exports[`heading() should display styles for size 2 1`] = `
.c0 {
  font-size: 37px;
  line-height: 1.2972972972972974;
  font-weight: semi-bold;
  font-family: 'Cabin',sans-serif;
}

<div
  className="c0"
/>
`;

exports[`heading() should display styles for size 3 1`] = `
.c0 {
  font-size: 28px;
  line-height: 1.2857142857142858;
  font-weight: semi-bold;
  font-family: 'Cabin',sans-serif;
}

<div
  className="c0"
/>
`;

exports[`heading() should display styles for size 4 1`] = `
.c0 {
  font-size: 22px;
  line-height: 1.2727272727272727;
  font-weight: semi-bold;
  font-family: 'Cabin',sans-serif;
}

<div
  className="c0"
/>
`;

exports[`heading() should display styles for size 5 1`] = `
.c0 {
  font-size: 18px;
  line-height: 1.3333333333333333;
  font-weight: semi-bold;
  font-family: 'Cabin',sans-serif;
}

<div
  className="c0"
/>
`;

exports[`heading() should display styles for size 6 1`] = `
.c0 {
  font-size: 15;
  line-height: 1.3333333333333333;
  font-weight: semi-bold;
  font-family: 'Cabin',sans-serif;
}

<div
  className="c0"
/>
`;
